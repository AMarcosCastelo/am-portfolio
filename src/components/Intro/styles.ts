import styled from 'styled-components'
import { motion } from 'framer-motion'

export const Box = styled(motion.div)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);

  width: 65vw;
  height: 55vh;
  display: flex;

  background: linear-gradient(
        to right,
        ${({ theme }) => theme.palette.background.default} 50%,
        ${({ theme }) => theme.palette.text.main} 50%
      )
      bottom,
    linear-gradient(
        to right,
        ${({ theme }) => theme.palette.background.default} 50%,
        ${({ theme }) => theme.palette.text.main} 50%
      )
      top;
  background-repeat: no-repeat;
  background-size: 100% 2px;

  border-left: 2px solid ${({ theme }) => theme.palette.background.default};
  border-right: 2px solid ${({ theme }) => theme.palette.text.main};
  z-index: 1;
`

export const SubBox = styled.div`
  width: 50%;
  position: relative;
  display: flex;

  .pic {
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translate(-50%, 0%);
    width: 100%;
    height: auto;
    max-width: 20rem;
  }
`

export const Text = styled.div`
  font-size: calc(1em + 1.5vw);
  color: ${({ theme }) => theme.palette.background.default};
  padding: 2rem;
  cursor: pointer;

  display: flex;
  flex-direction: column;
  justify-content: space-evenly;

  & > * :last-child {
    color: ${({ theme }) =>
      `rgba(${theme.palette.background.defaultRgba}, 0.6)`};
    font-size: calc(0.5rem + 1.5vw);
    font-weight: 300;
  }
`
