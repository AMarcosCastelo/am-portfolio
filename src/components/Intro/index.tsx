import { motion } from 'framer-motion'

import * as S from './styles'

const Intro = () => (
  <S.Box
    initial={{ height: 0 }}
    animate={{ height: '55vh' }}
    transition={{ type: 'spring', duration: 2, delay: 1 }}
  >
    <S.SubBox>
      <S.Text>
        <h1>Hi,</h1>
        <h2>{"I'm Antonio Marcos."}</h2>
        <h6>{"I'm a front-end developer."}</h6>
      </S.Text>
    </S.SubBox>

    <S.SubBox>
      <motion.div
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        transition={{ duration: 1, delay: 2 }}
      >
        <img
          className="pic"
          src="/assets/images/profile-img.png"
          alt="Profile Pic"
        />
      </motion.div>
    </S.SubBox>
  </S.Box>
)

export default Intro
