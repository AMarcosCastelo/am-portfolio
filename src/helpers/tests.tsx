import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'

import { LIGHT_THEME, DARK_THEME } from 'styles'

export const renderWithTheme = (
  children: React.ReactNode,
  theme?: 'light' | 'dark'
): RenderResult =>
  render(
    <ThemeProvider theme={theme === 'dark' ? DARK_THEME : LIGHT_THEME}>
      {children}
    </ThemeProvider>
  )
