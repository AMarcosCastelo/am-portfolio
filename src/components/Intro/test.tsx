import { renderWithTheme } from 'helpers'

import Intro from '.'

describe('<Intro />', () => {
  it('should render the Intro', () => {
    const { container } = renderWithTheme(<Intro />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
