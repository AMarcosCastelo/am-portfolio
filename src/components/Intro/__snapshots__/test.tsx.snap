// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> should render the Intro 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  width: 65vw;
  height: 55vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: linear-gradient( to right,#FCF6F4 50%,#000 50% ) bottom,linear-gradient( to right,#FCF6F4 50%,#000 50% ) top;
  background-repeat: no-repeat;
  background-size: 100% 2px;
  border-left: 2px solid #FCF6F4;
  border-right: 2px solid #000;
  z-index: 1;
}

.c1 {
  width: 50%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .pic {
  position: absolute;
  bottom: 0;
  left: 50%;
  -webkit-transform: translate(-50%,0%);
  -ms-transform: translate(-50%,0%);
  transform: translate(-50%,0%);
  width: 100%;
  height: auto;
  max-width: 20rem;
}

.c2 {
  font-size: calc(1em + 1.5vw);
  color: #FCF6F4;
  padding: 2rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c2 > *:last-child {
  color: rgba(252,246,244,0.6);
  font-size: calc(0.5rem + 1.5vw);
  font-weight: 300;
}

<div
  class="c0"
  style="height: 0px;"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1>
        Hi,
      </h1>
      <h2>
        I'm Antonio Marcos.
      </h2>
      <h6>
        I'm a front-end developer.
      </h6>
    </div>
  </div>
  <div
    class="c1"
  >
    <div
      style="opacity: 0;"
    >
      <img
        alt="Profile Pic"
        class="pic"
        src="/assets/images/profile-img.png"
      />
    </div>
  </div>
</div>
`;
